sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -811,34 +811,21 @@ public void close() {
             queryOptions.setSelect(select);
         }
 
-        return implementation.getTables().queryEntitiesWithPartitionAndRowKeyWithResponseAsync(tableName, partitionKey,
+        return implementation.getTables().queryEntityWithPartitionAndRowKeyWithResponseAsync(tableName, partitionKey,
             rowKey, timeoutInt, null, queryOptions, context)
             .handle((response, sink) -> {
-                final TableEntityQueryResponse entityQueryResponse = response.getValue();
-                if (entityQueryResponse == null) {
-                    logger.info("TableEntityQueryResponse is null. Table: {}, partition key: {}, row key: {}.",
-                        tableName, partitionKey, rowKey);
+                final Map<String, Object> matchingEntity = response.getValue();
 
-                    sink.complete();
-                    return;
-                }
-                final List<Map<String, Object>> matchingEntities = entityQueryResponse.getValue();
-                if (matchingEntities == null || matchingEntities.isEmpty()) {
+                if (matchingEntity == null || matchingEntity.isEmpty()) {
```

Comment:
```suggestion
                if (matchingEntity == null) {
```

The isEmpty check was only because we got a list of (0..1) items. Now that we get only a single map, we should simply ensure it's not null.